src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -801,7 +801,7 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
           = co_await _id_allocator_frontend.local().allocate_id(timeout);
         if (pid_reply.ec != errc::success) {
             vlog(txlog.warn, "allocate_id failed with {}", pid_reply.ec);
-            co_return init_tm_tx_reply{tx_errc::invalid_txn_state};
+            co_return init_tm_tx_reply{tx_errc::not_coordinator};
```

Comment:
what motivated this change to retryable errors?

Reply:
when an operation wasn't performed it's safe to return not_coordinator to ask a client to retry; when a client receives invalid_txn_state it treats it as a fatal error and an app is forced to recreate a producer, it's bad ux